src/le.py

```diff
@@ -1167,10 +1167,10 @@ def config_formatters():
 
 def extract_token(log_):
     """Extract the log token value if it exists"""
-    try:
-        if 'log' in log_ and log_['log']['source_type'] is 'token':
-            return log_['log']['tokens'][0]
```

Comment:
John, the if clause above will throw an exception if 'log' or 'source_type' is not included in the config file. We can use the get method instead which will return None, if the parameters are not there. 

if log_.get('log').get('source_type') is 'token':
          token = log_.get('log').get('tokens')[0]

But......using a try-catch creates a robust environment because the tokens list could be empty (source _type may not be configured correctly) and this will throw an exception too. Lets put the try-catch back in there :)